app/src/main/java/com/github/studydistractor/sdp/distraction/FirebaseDistractionStat.kt

```diff
@@ -52,6 +55,25 @@ class FirebaseDistractionStat constructor(did: String): DistractionStatInterface
             }
         })
 
+        val likesRef = db.getReference(pathLikes).child(did)
+        likesRef.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                likes.value = snapshot.children.count()
+            }
+            override fun onCancelled(error: DatabaseError) {
+                Log.d("Firebase", "loadPost:onCancelled " + error.toException().toString())
+            }
+        })
+        val dislikesRef = db.getReference(pathDislikes).child(did)
+        dislikesRef.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                dislikes.value = snapshot.children.count()
+            }
+            override fun onCancelled(error: DatabaseError) {
+                Log.d("Firebase", "loadPost:onCancelled " + error.toException().toString())
+            }
+        })
```

Comment:
Yes I also think about it but the issue is that in the onDataChange function the check and the transformation is not always the same. I should be great but I think we don't gain a lot of character if we need to pass always the function method in it. But it will be needed to talk about it with the others.